### ISSUE TITLE
enhance documentation (wiki)

### ISSUE THREAD

**Issue by dbuscombe-usgs** [Doodleverse/segmentation_gym#57]
needs additional wiki pages for (at least)

- [x] model evaluation
- [x] single model deployment
- [x] ensemble model deployment
- [ ] troubleshooting

**Comment by dbuscombe-usgs** [comment]
Dan makes a new case study for ensemble models

**Comment by ebgoldstein** [comment]
ordering: https://stackoverflow.com/questions/39396362/github-wiki-pages-sort-order

**Comment by 2320sharon** [comment]
# Questions About the Wiki
## Models
1. [standard vs spatial dropout](https://github.com/Doodleverse/segmentation_gym/wiki/Models-in-Gym)
- When should the user consider using Standard vs Spatial Dropout? I think providing some general tips on when the use which dropout type would be insightful.
2. Using [seg_images_in_folder.py](https://github.com/Doodleverse/segmentation_gym/blob/main/seg_images_in_folder.py)
- When I run this `seg_images_in_folder.py` with a pre-trained model with two sets of `.h5` weights one labeled as `full model` and the other with the standard model name should I use both sets of weights or just one? If I am supposed to use only one set of weights which one should I use for running predictions on sample imagery in inference mode?
```
/weights
  |__ sample_model_name.h5
  |__ sample_model_name_full_model.h5
```

**Comment by dbuscombe-usgs** [comment]
Good points, both.

1. is easy - we have no idea. Its there if someone wants to use it. I have never used it. Folks can look up in the keras docs what the difference is, but agreed that we should make a note of it

2. Agree this is confusing. You could use either set of  .h5 model weights. The "fullmodel.h5" version is for serving models through Zoo (needs a config file with the same file root).

**Comment by 2320sharon** [comment]
Ah that makes sense. I used both `.h5` files when I ran `seg_images_in_folder.py` and it somehow still worked. 
I'll update the wiki to state the purpose of the `fullmodel.h5`.

**Comment by dbuscombe-usgs** [comment]
I just made some improvements to the wiki that cover some of these themes, but I'll leave this open so we can add text for

- [ ] model evaluation
- [ ] explanation of `fullmodel`
- [ ] Gym utils
- [ ] troubleshooting

**Comment by 2320sharon** [comment]
I'll try to contribute to some of these topics when I can. Thanks for writing these out as a task list 👍

**Comment by dbuscombe-usgs** [comment]
add docs for loss weighting using LOSS_WEIGHTS in config file

- [ ] behaviour when true
- [ ] behaviour when false
- [ ] behaviour when a list of weights are passed

Only applied when Dice loss is used

**Comment by ebgoldstein** [comment]
folding in #89 : "provide documentation on end-of-training metrics - how they should be interpreted and used"

**Comment by dbuscombe-usgs** [comment]
Added some edits to most of the wiki pages, including an updated description of the config files, and a new page describing utilities: https://github.com/Doodleverse/segmentation_gym/wiki/9_Gym_utilities

The remaining items are new pages for 

- [ ] troubleshooting tips - I'm happy to take this on. But what should this include? Ideas below ⬇️ 
- [ ] use of loss weighting - @ebgoldstein you want to tackle this one?
- [ ] model evaluation - this should reflect the idea the model should be tested against an independent test dataset, one that ideally is composed of samples that are different from those used in training/validation, for example different weather, locations, etc (i.e. out-of-distribution data)

**Comment by dbuscombe-usgs** [comment]
Ah, docs.... the final frontier

**Comment by dbuscombe-usgs** [comment]
I updated the wiki, adding details about LOSS_WEIGHTS

Model evaluation will change when we implement planned changes to data splits

I  created a new page in the wiki for troubleshooting and organized it by heading. Will open a new issue now soliciting ideas

closing
